src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -52,9 +54,18 @@
   }
 
   private final boolean useCompactMapEntries;
+  private final BehaviorOnNullValues behaviorOnNullValues;
 
-  public DataConverter(boolean useCompactMapEntries) {
+  /**
+   * @param behaviorOnNullValues How to treat records with null values. Cannot be null; if in doubt,
+   * {@link BehaviorOnNullValues#DEFAULT} can be used.
+   */
+  public DataConverter(boolean useCompactMapEntries, BehaviorOnNullValues behaviorOnNullValues) {
     this.useCompactMapEntries = useCompactMapEntries;
+    this.behaviorOnNullValues = Objects.requireNonNull(
+        behaviorOnNullValues,
+        "behaviorOnNullValues cannot be null. Possible fix: use BehaviorOnNullValues.DEFAULT instead."
```

Comment:
Wouldn't making the Checkstyle build pass with the 4.0.x branch's standards mean fixing every Checkstyle error in every file, not just the errors present in the changes I made?

Reply:
The lines you are not adding have been fixed in `4.0.x`. 

The goal is make the new lines pass the checks (I've given several examples where they won't) and also minimize conflicts from 3.3.x to 4.0.x

Reply:
The `4.0.x` branch as it currently stands should already pass the checkstyle rules. So temporarily rebasing against this branch (if you can do this) would apply your changes on top of what's currently in `4.0.x`, so checkstyle will make sure that _your changes_ (plus all non-changed lines) satisfy the rules. When you take these commits and then rebase them on top of `3.3.x`, your changes should satisfy the 4.0.x checkstyle rules.

This may not be that straightforward, though, and it may depend on how cleanly git merge can do the rebase. You'd have to fix the checkstyle rules anyway for the `4.0.x` branch as a separate merge process after this PR is merged to the `3.3.x` branch, so IMO it's worth a shot doing it now.

Reply:
There are also features on the `4.0.x` branch that aren't included on the `3.3.x` line (e.g., `drop.invalid.message`; should those be merged in as well?